### PR TITLE
Add causal variant in fa benchmark

### PR DESCRIPTION
**Pull request by ESI-SYD**
Releated: https://github.com/intel/intel-xpu-backend-for-triton/issues/2179

CI: https://github.com/intel/intel-xpu-backend-for-triton/actions/runs/11047304993

### CODE REVIEW

**Review by whitneywhtsang — COMMENTED** [review]
can we add causal to the x_vals?

**Comment by ESI-SYD** [comment]
> can we add causal to the x_vals?

Yes, then we will merge the data with `causal=True` and `causal=False` to one report csv?

**Comment by whitneywhtsang** [comment]
> > can we add causal to the x_vals?
> 
> Yes, then we will merge the data with `causal=True` and `causal=False` to one report csv?

Right, WDYT?

**Comment by ESI-SYD** [comment]
> > > can we add causal to the x_vals?
> > 
> > 
> > Yes, then we will merge the data with `causal=True` and `causal=False` to one report csv?
> 
> Right, WDYT?

OK

File: benchmarks/triton_kernels_benchmark/flash_attention_fwd_benchmark.py [hunk 1]
```diff
@@ -226,25 +237,31 @@ def benchmark(Z, H, N_CTX, D_HEAD, provider):
...
-                                                                     False, scale=sm_scale), warmup=10, rep=10,
+                                                                     CAUSAL, scale=sm_scale), warmup=10, rep=10,
             quantiles=quantiles, fast_flush=False)
 
     elif provider == 'triton':
-        triton_fn = lambda: forward(q, k, v, causal, sm_scale)
-        if benchmark_suit.USE_IPEX_OPTION:
-            torch_fn = lambda: torch.nn.functional.scaled_dot_product_attention(
-                q, k, v, attn_mask=None, dropout_p=0.0, is_causal=False, scale=sm_scale).to(torch.float32)
+        # FIXME: remove bellow if condition when extend attention support for Causal = True done
```
Comment (whitneywhtsang) [inline benchmarks/triton_kernels_benchmark/flash_attention_fwd_benchmark.py:244]:
```suggestion
        # FIXME: remove below if condition when extend attention support for Causal = True done
```

**Comment by whitneywhtsang** [comment]
@Egor-Krivov With this change, can we still compare the results with the reported data? The reported data doesn't have the causal field, but it should be considered as false.

**Comment by whitneywhtsang** [comment]
@Egor-Krivov FYI, this PR is now merged.

**Comment by whitneywhtsang** [comment]
> > @Egor-Krivov FYI, this PR is now merged.
> 
> Dashboards should work now

Thanks Egor!

**Pull request merged**